src/build/patches.ts

```diff
@@ -0,0 +1,70 @@
+import { parse } from "kdljs";
+import { Enum } from "./types";
+import { readdir, readFile } from "fs/promises";
+import { merge } from "./helpers.js";
+
+/**
+ * Converts patch files in KDL to match the [types](types.d.ts).
+ */
+export function parseKDL(kdlText: string) {
+  const { output, errors } = parse(kdlText);
+
+  if (errors.length) {
+    throw new Error("KDL parse errors", { cause: errors });
+  }
+
+  const nodes = output!;
+  const enums: Record<string, Enum> = {};
+
+  for (const node of nodes) {
+    if (node.name === "enum") {
+      // Handle enum
+      const name = node.values[0];
```

Comment:
This was buried in older comments somehow, but would be nicer to extract this enum parsing code to a separate function, as we plan to add more things here. Otherwise parseKDL will become way too large to read.